src/oauth2.js

```diff
@@ -105,10 +106,24 @@ module.exports = function(config)
 						const grant = {
 							grant_type: "authorization_code",
         					code: params.code,
-							app_id: params.app_id
+							app_id: params.app_id,
+							workspace: params.workspace
 						};
 
 						const data = await getToken(grant);
+						if (params.workspace) {
+							data.additional_params = {
+								...data.additional_params,
+								workspace: params.workspace
+							};
```

Comment:
@MuhammadAliGhaffarr No need for reassignment here. Simply define `data.additional_params.workspace`